Library/Homebrew/cmd/update-report.rb

```diff
@@ -133,6 +132,8 @@ def output_update_report
         "git", "-C", HOMEBREW_REPOSITORY, "tag", "--list", "--sort=-version:refname", "*.*"
       ).lines.first.chomp
 
+      puts "Updated Homebrew from #{old_tag} (#{shorten_revision(initial_revision)}) "\
+           "to #{new_tag} (#{shorten_revision(current_revision)})."
       new_repository_version = new_tag if new_tag != old_tag
```

Comment:
Needs to only output `old_tag` and `new_tag` if they aren't the same. Also, I'm wondering if this will be confusing for folks where `Homebrew::EnvConfig.update_to_tag?` is not true and they are updated between commits on `master` instead of tags. I'd suggest in both of those cases the `old_tag` and `new_tag` is just omitted in favour of outputting revisions.

Also, feels like some overlap with https://github.com/Homebrew/brew/blob/c2ac73c04fbed880bee0aa9a444689d86c281027/Library/Homebrew/cmd/update-report.rb#L253-L265

Reply:
After getting into the code, it is already confusing even for me. I do not know why `update_to_tag` is not set for my repo, because I was pretty sure I am updating between tags. Maybe because of the bottom message with tags. The bottom message doesn't appear when there is no tag caught between updated commit range.

I prefer to know if I am on updating between tags, or just fetching the latest HEAD. Even with HEAD it is interesting to see if there were any release in between.  Something like this.
````
Updated Homebrew HEAD 5ee14b97f . . . 3.3.11 . cf4848b17
(+6 commits, use `git ...` to view them)
(+1 tag, see the release notes below)
  https://github.com/Homebrew/brew/releases/tag/3.3.11
```

Reply:
Whether you update between tags or between HEAD commits depends on whether you have developer mode enabled (`brew developer`).

Reply:
Yep, it reports it is enabled. I probably did it at some point. Still it would be nice to get a reminder that I am using development version.